internal/server/response_buffer_middleware.go

```diff
@@ -72,10 +74,27 @@ func (w *bufferedResponseWriter) WriteHeader(statusCode int) {
 	if !w.headerWritten {
 		w.statusCode = statusCode
 		w.headerWritten = true
+
+		if w.ShouldSwitchToUnbuffered() {
+			w.SwitchToUnbuffered()
+		}
 	}
 }
 
+func (w *bufferedResponseWriter) ShouldSwitchToUnbuffered() bool {
+	return w.Header().Get("Content-Type") == "text/event-stream"
```

Comment:
Emm, so what happens if the header is `Content-Type: text/event-stream; charset=UTF-8`? Seems like a bug?

Reply:
This should be:

```go
return strings.HasPrefix(w.Header().Get("Content-Type"), "text/event-stream")
```

Reply:
Oh, I see this is fixed in master.